src/App/Security/Authorization/Voter/PrivateDiscussionVoter.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Security\Authorization\Voter;
+
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Doctrine\Bundle\DoctrineBundle\Registry as Doctrine;
+
+class PrivateDiscussionVoter implements VoterInterface
+{
+    const VIEW       = 'view';
+    const EDIT       = 'edit';
+    const ADD_MEMBER = 'add_member';
+    const UNREADER   = 'unreader';
+
+    private $doctrine;
+
+    public function __construct(Doctrine $doctrine)
+    {
+        $this->doctrine = $doctrine;
+    }
+
+    public function supportsAttribute($attribute)
+    {
+        return in_array($attribute, array(
+            self::VIEW,
+            self::EDIT,
+            self::ADD_MEMBER,
+            self::UNREADER,
+        ));
+    }
+
+    public function supportsClass($class)
+    {
+        $supportedClass = 'App\Entity\PrivateDiscussion';
+
+        return $supportedClass === $class || is_subclass_of($class, $supportedClass);
+    }
+
+    public function vote(TokenInterface $token, $discussion, array $attributes)
+    {
+        // check if class of this object is supported by this voter
```

Comment:
If the voter does not support the object, the vote method will never be called, so this test is useless.